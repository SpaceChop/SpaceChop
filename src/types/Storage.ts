import * as t from 'runtypes';
import S3StorageConfig from '../storage/s3/types';

const Storage = t.Union(
  t.Record({ s3: S3StorageConfig }),
);

export type Storage = t.Static<typeof Storage>;
export default Storage;
