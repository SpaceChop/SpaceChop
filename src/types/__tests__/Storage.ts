import Storage from '../Storage';

describe('validate Storage', () => {
  describe('invalid', () => {
    it('should require valid storage array', () => {
      const storage = {};
      expect(() => Storage.check(storage)).toThrowErrorMatchingSnapshot();
    });

    it('should require valid storage', () => {
      const storage = [{}];
      expect(() => Storage.check(storage)).toThrowErrorMatchingSnapshot();
    });

    it('should throw on any other type of storage', () => {
      const storage = [{ bad_storage: {} }];
      expect(() => Storage.check(storage)).toThrowErrorMatchingSnapshot();
    });
  });
});
