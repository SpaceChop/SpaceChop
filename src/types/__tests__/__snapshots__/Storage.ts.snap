// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Storage invalid should require valid storage 1`] = `"Expected { s3: { access_key_id: string; secret_access_key: string; region: string; bucket_name: string; path: string; } & { ACL?: \\"private\\" | \\"public-read\\" | \\"public-read-write\\" | \\"authenticated-read\\" | \\"aws-exec-read\\" | \\"bucket-owner-read\\" | \\"bucket-owner-full-control\\"; endpoint?: string; }; }, but was object"`;

exports[`validate Storage invalid should require valid storage array 1`] = `"Expected { s3: { access_key_id: string; secret_access_key: string; region: string; bucket_name: string; path: string; } & { ACL?: \\"private\\" | \\"public-read\\" | \\"public-read-write\\" | \\"authenticated-read\\" | \\"aws-exec-read\\" | \\"bucket-owner-read\\" | \\"bucket-owner-full-control\\"; endpoint?: string; }; }, but was object"`;

exports[`validate Storage invalid should throw on any other type of storage 1`] = `"Expected { s3: { access_key_id: string; secret_access_key: string; region: string; bucket_name: string; path: string; } & { ACL?: \\"private\\" | \\"public-read\\" | \\"public-read-write\\" | \\"authenticated-read\\" | \\"aws-exec-read\\" | \\"bucket-owner-read\\" | \\"bucket-owner-full-control\\"; endpoint?: string; }; }, but was object"`;
